os-observer/src/utils/getPath.ts

```diff
@@ -0,0 +1,49 @@
+// ex: getPath({ user: { name: 'u1' }}, "user.name")
+export function getPath<
```

Comment:
Nooooo you're ruining my fun with typescript magic!

You make a good point and I will include that in a future PR. Because resulting object is typed the chaining doesn't need to be optional. This is a much better idea.